File: tarea_1/1.4_matrices/cpp/src/RedistribucionController.cpp
#include "../include/RedistribucionController.h"

using namespace std;

// Constructor: Inyección de dependencia del servicio de redistribución
RedistribucionController::RedistribucionController(unique_ptr<RedistribucionService> srv)
    : service(move(srv)) {}

// Ejecuta algoritmo de optimización greedy para redistribución
void RedistribucionController::ejecutarOptimizacion() {
    service->ejecutarOptimizacion();
}

// Muestra resultados de la optimización con costos y movimientos
void RedistribucionController::mostrarResultados() {
    service->mostrarResultados();
}

// Carga datos de prueba predefinidos para demostración
void RedistribucionController::cargarDatosPrueba() {
    service->cargarDatosPrueba();
}

// Permite configuración manual de matrices de entrada
void RedistribucionController::configurarMatrices() {
    service->configurarMatrices();
}

File: tarea_1/1.4_matrices/python/redistribucion_controller.py
from redistribucion_service import RedistribucionService

class RedistribucionController:
    def __init__(self, service: RedistribucionService):
        self.service = service
    
    def ejecutar_optimizacion(self) -> None:
        self.service.ejecutar_optimizacion()
    
    def mostrar_resultados(self) -> None:
        self.service.mostrar_resultados()
    
    def cargar_datos_prueba(self) -> None:
        self.service.cargar_datos_prueba()
    
    def configurar_matrices(self) -> None:
        self.service.configurar_matrices()